airflow/models/dag.py

```diff
@@ -896,7 +897,7 @@ def get_num_active_runs(self, external_trigger=None, session=None):
         )
 
         if external_trigger is not None:
-            query = query.filter(DagRun.external_trigger == external_trigger)
+            query = query.filter(DagRun.external_trigger == expression.literal(external_trigger))
```

Comment:
Not sure about this one -- feels a bit risky.

```suggestion
            query = query.filter(DagRun.external_trigger == (expression.true() if external_trigger else expression.false())))
```

feels less prone to risk of SQLi attack

Reply:
done